File: apps/frontend/src/types/mfa.ts
export interface MfaStatus {
  enrolled: boolean
  enrolled_at?: string
}

export interface MfaSetupResponse {
  qr_uri: string
  secret: string
}

export interface MfaVerifyResponse {
  valid: boolean
  session_token?: string
  expires_in?: number
}

export interface MfaUserStatus {
  user_id: string
  email: string
  first_name: string
  last_name: string
  role: string
  enrolled: boolean
  enrolled_at?: string
  last_verified_at?: string
}

export interface OnboardingDocuments {
  onboarding_id: string
  status: string
  id_front_url?: string
  id_back_url?: string
  selfie_url?: string
  signature_url?: string
  signed_at?: string
  id_type?: string
  id_number?: string
  first_name?: string
  last_name?: string
  date_of_birth?: string
  phone?: string
  emergency_contact_name?: string
  emergency_contact_phone?: string
}

File: apps/backend/app/schemas/mfa.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class MfaStatusResponse(BaseModel):
    enrolled: bool
    enrolled_at: Optional[datetime] = None


class MfaSetupResponse(BaseModel):
    qr_uri: str    # otpauth:// URI — frontend renders as QR code
    secret: str    # base32 secret for manual entry


class MfaConfirmRequest(BaseModel):
    code: str = Field(min_length=6, max_length=6)


class MfaVerifyRequest(BaseModel):
    code: str = Field(min_length=6, max_length=6)


class MfaVerifyResponse(BaseModel):
    valid: bool
    session_token: Optional[str] = None   # stored in Redis; valid for mfa_session_ttl_seconds
    expires_in: Optional[int] = None      # seconds until expiry


class MfaUserStatusResponse(BaseModel):
    user_id: str
    email: str
    first_name: str
    last_name: str
    role: str
    enrolled: bool
    enrolled_at: Optional[datetime] = None
    last_verified_at: Optional[datetime] = None


class MfaRevokeRequest(BaseModel):
    user_id: str
